src/classes/PermanentFileSystem.ts

```diff
@@ -381,6 +388,29 @@ export class PermanentFileSystem {
     );
   }
 
+  private async findArchiveRecordInParentDirectory(
+    parentPath: string,
+    archiveRecordName: string,
+    overrideParentCache = false,
+  ): Promise<ArchiveRecord> {
+    const parentFolder = await this.loadFolder(
+      parentPath,
+      overrideParentCache,
+    );
+    const targetArchiveRecord = parentFolder.archiveRecords.find(
+      (folder) => folder.fileSystemCompatibleName === archiveRecordName,
+    );
+
+    if (overrideParentCache && !targetArchiveRecord) {
+      throw new Error('The specified archive record does not exist');
+    }
+    return targetArchiveRecord ?? this.findArchiveRecordInParentDirectory(
```

Comment:
@slifty some notes on why this is recursive would help, thanks!

Reply:
Good call -- recursion deserves an explanation!  It's also probably the case that I should avoid recursion at all costs... but since I'm doing it i'll at least add the commment :D

Reply:
I also was being too clever by half in this -- I've removed the `??` and made the logic flow more explicit / legible.